lib/validates_timeliness/converter.rb

```diff
@@ -10,7 +10,7 @@ def initialize(type:, format: nil, ignore_usec: false, time_zone_aware: false)
     end
 
     def type_cast_value(value)
-      return nil if value.nil? || !value.respond_to?(:to_time)
+      return nil if value.nil? || !value.respond_to?(:to_time) || value.is_a?(Numeric)
```

Comment:
Can you add a failing spec?

At the moment I do not understand how it is possible that a `Numeric` value passes `!value.respond_to?(:to_time)`

Reply:
@tagliala  👇 
```
pry(main)> "444.0".respond_to?(:to_time)
=> true
pry(main)> 444.0.respond_to?(:to_time)
=> false
```

Reply:
Which would mean this guard would still fail, because the value is actually a string.
`>> "444.0".is_a?(Numeric)
 false
`

Reply:
But as @tagliala said. we'll need a failing spec for this. It is not clear how this occurs.

Reply:
The only thing that I can think of is that `Numeric` has been enhanced with `to_time` by another library, but that same gem does not define `to_date`. However, even in this use case that would fail the next line when methods like `acts_like?(:time)` are invoked

Reply:
@adzap @tagliala
The condition in [line #13](https://github.com/adzap/validates_timeliness/blob/beaca8685bd4a59fda5c7646cd5b9a2f754afc75/lib/validates_timeliness/converter.rb#L13) just check whether the passed value responds to `"to_time"` method or not.
whereas the issue is in [line #20](https://github.com/adzap/validates_timeliness/blob/beaca8685bd4a59fda5c7646cd5b9a2f754afc75/lib/validates_timeliness/converter.rb#L20) while calling `"to_date"` method not `"to_time"`!

The `Numeric` values are responds to `"to_time"` but not `"to_date"`.

Reply:
**Example:**
`# app/models/item.rb`
```ruby
validates_date :expire_date
```

`# console:`
```ruby
> item = Item.new(expire_date: 444)
> item.valid?
NoMethodError: undefined method `to_date' for 444:Integer
```

Reply:
Hi, thanks for the clarification.

> The `Numeric` values are responds to `"to_time"` but not `"to_date"`.

This is the part that does not sound.

Is there any third party library defining `to_time` on Numeric elements? Can you check where `to_time` is defined?

I've setup a pretty basic application at https://github.com/diowa/ruby3-rails7-bootstrap-heroku/tree/validates-timeliness

```
> 444.respond_to?(:to_time)
=> false
[2] pry(main)> Item.new(expire_date: 444).valid?
=> false
```

However... @adzap probably there could be a better detection if the `timeliness` helpers are available on the given value